fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PermissiveAvailability.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+import org.joda.time.Interval;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * An availability which doesn't force immutability on its contents.
+ */
+public class PermissiveAvailability implements Availability {
+
+    private final TableName name;
+    private final Set<Column> columns;
+    private final DataSourceMetadataService metadataService;
+
+    /**
+     * Constructor.
+     *
+     * @param tableName The name of the data source
+     * @param columns A set of columns associated with the data source
+     * @param metadataService A service containing the data source segment data
+     */
+    public PermissiveAvailability(
+            TableName tableName,
+            @NotNull Set<Column> columns,
+            @NotNull DataSourceMetadataService metadataService
+    ) {
+        this.name = tableName;
+        this.columns = columns;
+        this.metadataService = metadataService;
+    }
+
+    @Override
+    public Set<TableName> getDataSourceNames() {
+        return Collections.singleton(name);
```

Comment:
Rather than make this wrapping every time `getDataSourceNames` is called, we should do it only once in the constructor and store the collection.